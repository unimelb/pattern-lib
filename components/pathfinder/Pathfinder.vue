<template>
  <section
    :style="{ backgroundImage: `url(${image})` }"
    class="section section--image bg-inverted-dark">
    <div class="section--image-mask" />
    <div class="section__inner">
      <div class="listing listing--three listing--center">
        <ListItem
          v-for="box in boxes.split(',')"
          :key="box">
          <CardPathfinder
            href="www.unimelb.edu.au"
            :title="box">
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Natus facilis libero in, ipsam quidem cupiditate! Dignissimos pariatur, repellendus dolore nemo saepe, et dolor maxime, enim rem sapiente debitis minima blanditiis.
          </CardPathfinder>
        </ListItem>
      </div>
    </div>
  </section>
</template>

<script>
import CardPathfinder from '../cards/CardPathfinder.vue';

export default {
  name: 'Pathfinder',
  components: { CardPathfinder },
  props: {
    image: {
      type: String,
      default: 'https://via.placeholder.com/400x200',
    },
    boxes: {
      type: String,
      default: '',
    },
  },
};
</script>
