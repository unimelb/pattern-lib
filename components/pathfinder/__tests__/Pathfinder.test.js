import {
  shallow,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import Pathfinder from '../Pathfinder.vue';

expect.extend(toHaveNoViolations);

describe('Pathfinder', () => {
  it('should match snapshot', () => {
    const result = shallow(Pathfinder).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = shallow(Pathfinder).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
