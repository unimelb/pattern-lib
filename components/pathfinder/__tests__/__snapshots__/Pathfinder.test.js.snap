// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pathfinder should match snapshot 1`] = `
<section
  class="section section--image bg-inverted-dark"
  style="background-image: url(https://via.placeholder.com/400x200);"
>
  <div
    class="section--image-mask"
  />
   
  <div
    class="section__inner"
  >
    <div
      class="listing listing--three listing--center"
    >
      <listitem>
        <!---->
      </listitem>
    </div>
  </div>
</section>
`;
